appengine/test/tc_sample_apps.rb

```diff
@@ -32,7 +32,7 @@ def run_app_test(dirname)
     ::Dir.chdir(::File.join(APPS_DIR, dirname)) do |dir|
       build_docker_image("", dirname) do |image|
         run_docker_daemon("-p 8080:8080 #{image}") do |container|
-          assert_cmd_output("curl -s -S http://127.0.0.1:8080/", "ruby app", 10)
+          assert_cmd_output("docker run --link #{container}:#{container} gcr.io/google-appengine/debian8 /bin/bash -c 'apt-get -qq update > /dev/null 2>&1; apt-get -qq -y install curl > /dev/null 2>&1; curl -s -S #{container}:8080/'", "ruby app", 15)
```

Comment:
Ooh, this is a little annoying, did you try to run this in the ruby image itself instead? That way we could avoid linking and having to install curl.